examples/rllib.py

```diff
@@ -62,17 +97,22 @@ def main(
     checkpoint_num,
     save_model_path,
 ):
+    assert train_batch_size > 0, f"{train_batch_size.__name__} cannot be less than 1."
+    if rollout_fragment_length > train_batch_size:
+        rollout_fragment_length = train_batch_size
+
     pbt = PopulationBasedTraining(
         time_attr="time_total_s",
         metric="episode_reward_mean",
         mode="max",
         perturbation_interval=300,
         resample_probability=0.25,
         # Specifies the mutations of these hyperparams
+        # See: `ray.rllib.agents.trainer.COMMON_CONFIG` for common hyperparams
         hyperparam_mutations={
             "lr": [1e-3, 5e-4, 1e-4, 5e-5, 1e-5],
-            "rollout_fragment_length": lambda: random.randint(128, 16384),
-            "train_batch_size": lambda: random.randint(2000, 160000),
+            "rollout_fragment_length": lambda: rollout_fragment_length,
+            "train_batch_size": lambda: train_batch_size,
```

Comment:
These two lines were the problem since they used random values. The core of it was that the `ray.tune()` `stop` parameter was only being checked per training batch so with extremely large training batches it would get checked very infrequently.